docs/sources/reference/cli/environment-variables.md

```diff
@@ -0,0 +1,37 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/cli/environment-variables/
+description: Learn about the environment variables you can use with Alloy
+menuTitle: Environment variables
+title: Environment variables
+weight: 500
+---
+
+# Environment variables
+
+You can use environment variables to control the run-time behavior of {{< param "FULL_PRODUCT_NAME" >}}.
+
+The following environment variables are supported:
+
+* `GODEBUG`
+* `HTTP_PROXY`
+
+Refer to the [Go runtime][runtime] documentation for more information about Go runtime environment variables.
+
+## Arguments
+
+The following arguments are supported.
+
+### GODEBUG
+
+Argument               | Description                                                                                          | Default
+-----------------------|------------------------------------------------------------------------------------------------------|--------
+`x509usefallbackroots` | Enforce a fallback on the X.509 trusted root certificates. Set to `1` to enable.                     | `0`
+`netdns`               | Force a resolver. Set to `go` for a pure Go resolver. Set to `cgo` or `win32` for a native resolver. |
```

Comment:
I don't think this env var makes a difference in Alloy on Linux, because Alloy is built for Linux using a `netdns` flag. This forces the resolver to always be `netdns`. It might make more difference for windows though. Maybe this is worth documenting? cc @rfratto 

It may be worth documenting `GODEBUG=netdns=1` and `GODEBUG=netdns=2` though. they print extra information to logs. "2" is more verbose than "1".

Reply:
The build tag can be overridden at runtime. Looking at https://github.com/golang/go/blob/5a18e79687dea15680ff5f799b549fa0efd0cad9/src/net/conf.go#L86 there are several paths that can change the buildtag behavior by adjusting the prefer flags.

Reply:
This setting `GODEBUG=netdns=go` forces using go on my linux box. Specifically pivots on conf.d:235, whereas the default uses cgo resolver.

Reply:
I've added what's being discussed here and in the Slack thread. It's not "right" yet... just added as-is to capture the suggestions.